src/cocotb_tools/check_results.py

```diff
@@ -28,17 +28,27 @@ def get_results(results_xml_file: Path) -> tuple[int, int]:
             f"ERROR: Simulation terminated abnormally. Results file {results_xml_file} not found."
         )
 
+    # pytest --junit-xml=<file> is generating proper JUnit XML report file
+    # It is using errors attribute as indicator for failed tests without execution but
+    # also as indicator for failed pytest (invalid arguments, configuration, setup, ...)
     num_tests = 0
-    num_failed = 0
+    num_failed = 0  # Failed tests during execution (including setup and call)
+    num_errors = 0  # Errors in pytest configuration, fixtures, setup, tests
 
     tree = ElementTree.parse(results_xml_file)
     for ts in tree.iter("testsuite"):
-        for tc in ts.iter("testcase"):
-            num_tests += 1
-            for _ in tc.iter("failure"):
-                num_failed += 1
-
-    return (num_tests, num_failed)
+        if "tests" in ts.attrib:  # pytest, compatible with JUnit XML specification
+            num_tests += int(ts.attrib.get("tests", 0))
+            num_failed += int(ts.attrib.get("failures", 0))
+            num_errors += int(ts.attrib.get("errors", 0))
+
+        else:  # cocotb, non-compatible with Junit XML specification
+            for tc in ts.iter("testcase"):
+                num_tests += 1
+                for _ in tc.iter("failure"):
+                    num_failed += 1
```

Comment:
Can we just make cocotb be JUnit compatible? There is an open issue against this issue. If you want, that can be a separate PR.

Reply:
@ktbarrett I have added a `TODO` comment in this commit fd38a3163eac81e8157eede94a6d299454c327b7 

`XUnitReporter` and part of code in cocotb `regression.py` that is using this class requires some re-work.

I have noticed this opened Pull Request https://github.com/cocotb/cocotb/pull/4854 that is trying to fix that and related issue https://github.com/cocotb/cocotb/issues/4171 I can help here but I would do that in little different way that original PR. For example, I wouldn't magically move attributes to properties etc.

Reply:
#5110 should make this possible.